### PR TITLE
Use ApolloClient as a basic store/state management 

### PR DESCRIPTION
**Pull request by cdevent**
### Use ApolloClient 3 [reactive variables](https://www.apollographql.com/docs/react/local-state/reactive-variables/) as a basic store/state management.  

In general, we need to remove the unnecessary redux state management package as we have a nice alternative as Apollo Client and from version 3 they have a nice solution for that as **reactive variables**. So the main idea was to use Apollo Client 3 [reactive variables in tandem with hooks](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/#storing-local-state-in-reactive-variables), created a basic `/src/storage/store.ts` file where we can define our state and its initial state and create hooks for each statement and get it whenever I need it in components. 

For example, we create and export reactive variable for the facilities:  

`export const facilities = makeVar<FacilitiesState>({ current: null, list: [] });`

Then create a custom hook that exports the current statement and do any operations related to this var: 

`/src/hooks/useFacilities.ts`

Now U can use hook `useFacilities()` whenever you want to access to current selected Facility or change it to another: 

- get the currently selected facility: 

```

  const {
    facilities: { current },
  } = useFacilities();

```

Or, use general appConfig state with `useAppConfig()` hook:

get current plid in patientApp: 

`  const {config : {plid}} = useAppConfig();  `
  
get current organization : 
```
  const {
    config: { organization },
  } = useAppConfig();

```
to update any config field we can add any method into the `useAppConfig() ` hook or use general method  `updateConfigField(field: string, value: any)
`

## Additional Information

Fixes https://github.com/CDCgov/prime-simplereport/issues/1448

TODO: 
- [x] Remove React-Redux functions and store from:
  - [x] PatientApp
  - [x] SimpleReport

- [x] Rewrite all tests where we are used redux store.
- [x] Fix usage in stories.
- [ ] Check and Add more types and typechecks especially for hooks.  

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### CODE REVIEW

File: frontend/src/app/constants/constants.d.ts [hunk 1]
```diff
@@ -611,6 +611,18 @@ interface PersonFormData extends Person {
   facilityId: string | null;
 }
 
+interface Patient extends PersonFormData {
```
Comment (timothybest-usds) [inline frontend/src/app/constants/constants.d.ts:614]: I don't think this type definition is correct. The `Patient` in the AoE Form is similar but different then the one stored in the redux state for the patient experience. the two types come from different backend APIs

File: frontend/src/app/facilitySelect/WithFacility.tsx [hunk 2]
```diff
@@ -1,47 +1,63 @@
-import React from "react";
-import { useDispatch, useSelector, connect } from "react-redux";
+/* eslint-disable graphql/template-strings */
```
Comment (timothybest-usds) [inline frontend/src/app/facilitySelect/WithFacility.tsx:1]:
```suggestion
```
Why is this necessary? It doesn't look like there are any graphql template strings in this file

File: frontend/src/app/facilitySelect/WithFacility.tsx [hunk 3]
```diff
@@ -1,47 +1,63 @@
...
 import FacilitySelect from "./FacilitySelect";
 
-const Loading: React.FC<{}> = () => <p>Loading facility information...</p>;
+const Loading: FC<{}> = () => <p>Loading facility information...</p>;
 
 interface Props {}
 
-const WithFacility: React.FC<Props> = ({ children }) => {
-  const dispatch = useDispatch();
+const WithFacility: FC<Props> = ({ children }) => {
```
Comment (timothybest-usds) [inline frontend/src/app/facilitySelect/WithFacility.tsx:16]: This component no longer works. 

Steps to reproduce
1. Add a second facility via http://localhost:3000/settings/add-facility/
2. go to http://localhost:3000/
3. click on a facility
4. you will still see the facility select screen

Expected behavior
the facility is selected and you are routed to the app

File: frontend/src/app/Settings/types.d.ts [hunk 4]
```diff
@@ -183,3 +183,18 @@ interface FacilityData {
 }
 
 type TestCorrectionStatus = "ORIGINAL" | "CORRECTED" | "REMOVED";
+
+interface FacilitiesState {
+  current: Facility | null;
+  list: Facility[];
```
Comment (timothybest-usds) [inline frontend/src/app/Settings/types.d.ts:189]:
```suggestion
  selectedFacility: Facility | null;
  availableFacilities: Facility[];
```

`list` and `current` are ambiguous names that are hard to parse within the components.

**Review by timothybest-usds — CHANGES_REQUESTED** [review]

File: frontend/src/patientApp/PatientApp.tsx [hunk 5]
```diff
@@ -1,14 +1,15 @@
+//@ts-nocheck
```
Comment (timothybest-usds) [inline frontend/src/patientApp/PatientApp.tsx:1]:
```suggestion
```
This need to be removed before merging

File: frontend/src/storage/store.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    roleDescription: "",
+    isAdmin: false,
+  },
+  organization: {
+    name: "",
+  },
+  activationToken: null,
+});
+
+export const patient = makeVar<Patient>({
```
Comment (timothybest-usds) [inline frontend/src/storage/store.ts:24]: I am a little worried about the data implications of defaulting to a mostly empty patient. It seems that the patient in the store should be set to `null` when there is no patient

File: frontend/src/patientApp/timeOfTest/TestResult.tsx [hunk 7]
```diff
@@ -1,16 +1,20 @@
-import { useSelector } from "react-redux";
-
+//@ts-nocheck
```
Comment (timothybest-usds) [inline frontend/src/patientApp/timeOfTest/TestResult.tsx:1]:
```suggestion
```

needs to be removed before merge

**Review by timothybest-usds — CHANGES_REQUESTED** [review]

File: frontend/src/app/facilitySelect/WithFacility.tsx (hunk 2 above)
Comment (cdevent) [reply frontend/src/app/facilitySelect/WithFacility.tsx:1]: I think this can be removed, it was previously added.

File: frontend/src/app/facilitySelect/WithFacility.tsx (hunk 3 above)
Comment (cdevent) [reply frontend/src/app/facilitySelect/WithFacility.tsx:16]: With this I need some help, I can't add another facility. 
Screenshot 2021-06-08 at 7 39 48 PM

File: frontend/src/app/Settings/types.d.ts (hunk 4 above)
Comment (cdevent) [reply frontend/src/app/Settings/types.d.ts:189]: good point, thank you. will change.

File: frontend/src/app/telemetry-provider.jsx [hunk 8]
```diff
@@ -26,8 +26,9 @@ class TelemetryProvider extends Component {
       ai.initialize(AppInsightsInstrumentationKey, history);
       this.setState({ initialized: true });
     }
-
-    this.props.after();
+    if (after) {
```
[the remark this answers is not in the document]
Comment (cdevent) [reply frontend/src/app/telemetry-provider.jsx:29]: I think yes, but i thought we have already fix for that,  can you mention me this issue?

File: frontend/src/storage/store.ts (hunk 6 above)
Comment (cdevent) [reply frontend/src/storage/store.ts:24]: Good point.

File: frontend/src/app/Settings/types.d.ts [hunk 9]
```diff
@@ -183,3 +183,18 @@ interface FacilityData {
...
+}
+
+interface AppConfigState {
+  organization: {
+    name: string;
+  };
+  dataLoaded: Boolean;
```
[the remark this answers is not in the document]
Comment (cdevent) [reply frontend/src/app/Settings/types.d.ts:196]: oops, will change. sorry.

File: frontend/src/app/constants/constants.d.ts (hunk 1 above)
Comment (cdevent) [reply frontend/src/app/constants/constants.d.ts:614]: Yes, they come from different API that was a reason to add another extended version on the patientApp.

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 10]
```diff
@@ -414,7 +414,7 @@ const TestResultsList = (props: TestResultsListProps) => {
 
   const queryVariables: {
     patientId?: string;
-    facilityId: string;
+    facilityId: string | null | undefined;
```
[the remark this answers is not in the document]
Comment (cdevent) [reply frontend/src/app/testResults/TestResultsList.tsx:417]: got it, but it's look like we already throw an error and do not render the component if facility is not selected, line 446

File: frontend/src/storage/cache.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import { InMemoryCache } from "@apollo/client";
+
+import store from "./store";
+
+const cache: InMemoryCache = new InMemoryCache({
+  typePolicies: {
+    Query: {
+      fields: {
+        store: {
```
Comment (nullflux) [inline frontend/src/storage/cache.ts:9]: I think it's _okay_ to have a single InMemoryCache field for `store`, but I think that one of the most powerful features of Apollo's reactive variables is the ability to automatically update previously-made Queries based on a cached variable having been updated.

Taking advantage of this capability would require that we cache those query variables individually. I understand that we're not using it that way, here, but just wanted to call it out

File: frontend/src/patientApp/timeOfTest/PatientFormContainer.tsx [hunk 12]
```diff
@@ -89,9 +92,11 @@ const PatientFormContainer = () => {
             patient={{
               ...patient,
               facilityId:
-                patient.facility === null ? null : patient.facility?.id,
+                patient?.facility === null
+                  ? null
+                  : (patient?.facility?.id as string),
```
Comment (nullflux) [inline frontend/src/patientApp/timeOfTest/PatientFormContainer.tsx:97]: I think it's usually safer to cast to string in this way, rather than using `as`

```suggestion
                  : String(patient?.facility?.id),
```

Alternatively, could we fix the type of `patient.facility` to understand that `id` is in fact a string?

File: frontend/src/patientApp/PxpApiService.ts [hunk 13]
```diff
@@ -87,6 +87,7 @@ export class PxpApi {
     dateOfBirth: string,
     data: UpdatePatientData
   ) {
+    console.log(patientLinkId, dateOfBirth, data);
```
[the remark this answers is not in the document]
Comment (cdevent) [reply frontend/src/patientApp/PxpApiService.ts:90]: will be removed in my next commit, thank you! Just try to clean up all tests from redux, to allow them to pass now.

File: frontend/src/app/telemetry-provider.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (nullflux) [reply frontend/src/app/telemetry-provider.jsx:29]: I don't thiiiink this is related, but here's the issue Tim is talking about: https://github.com/CDCgov/prime-simplereport/pull/1711

File: frontend/src/patientApp/timeOfTest/PatientFormContainer.tsx (hunk 12 above)
Comment (cdevent) [reply frontend/src/patientApp/timeOfTest/PatientFormContainer.tsx:97]: I hope yes, it's in my list of fixes.
